test/mocks/server/mocks.h

```diff
@@ -471,6 +471,15 @@ class MockHealthCheckerFactoryContext : public virtual HealthCheckerFactoryConte
   testing::NiceMock<Envoy::Upstream::MockHealthCheckEventLogger>* event_logger_{};
 };
 
+class MockAdminStream : public Server::AdminStream {
+public:
+  MOCK_METHOD1(setEndStreamOnComplete, void(bool));
```

Comment:
Nit: mock classes should have constructors/destructors, which should be defined in the `.cc`. This has to do with some efficiency argument with gmock, see https://github.com/google/googletest/blob/master/googlemock/docs/CookBook.md#making-the-compilation-faster.